plone/app/upgrade/v52/final.py

```diff
@@ -238,3 +240,28 @@ def migrate_site_logo_from_ascii_to_bytes(context):
     you get a WrongType error when saving the site-controlpanel.
     """
     migrate_record_from_ascii_to_bytes("plone.site_logo", ISiteSchema, prefix="plone")
+
+
+def _recursive_strict_permission(obj):
+    obj.manage_permission(view, ('Manager', 'Owner'), 0)
+    if base_hasattr(obj, 'objectValues'):
+        for child in obj.objectValues():
+            _recursive_strict_permission(child)
+
+
+def secure_portal_setup_objects(context):
+    """Make portal_setup objects accessible only to Manager/Owner.
+
+    This matches the GenericSetup code for new logs and snapshots.
+    See https://github.com/zopefoundation/Products.GenericSetup/pull/102
+    """
+    # context conveniently is the portal_setup too.
+    # Set permission on setup tool and its sub objects, which are the logs.
+    _recursive_strict_permission(context)
```

Comment:
You don't need to set the permission on the tool itself. It's already protected by `security.declareObjectProtected(view_management_screens)` in the class.

Reply:
I wondered about that, and I now checked. There *is* a slight difference. If the object is protected, and the View permission is acquired, and the tool has a published method protected by View, then anonymous can view it.
Methods marked with `security.public` are available in both cases.
But there are no methods protected by View on the setup tool. And probably if there were, then they are meant to be viewed.
Also, this code would introduce a difference between a new site and an upgraded site.
I will revert the part of the PR that changes the tool itself.

Reply:
Just for general clarification, restricting `View`/removing acquisition on the tool will prevent  browsers accessing a log or snapshot via URL. It will not prevent the XML-RPC rendering. Doing the same for `Access contents information` will prevent the XML-RPC hole. I describe the manual fix at https://github.com/zopefoundation/Products.GenericSetup/security/advisories/GHSA-jff3-mwp3-f8cw